src/Admin/GraphiQL/GraphiQL.php

```diff
@@ -109,13 +109,13 @@ public function register_admin_page() {
 	}
 
 	/**
-	 * Render the markup to load GraphiQL to
+	 * Render the markup to load GraphiQL to.
 	 *
 	 * @return void
 	 */
 	public function render_graphiql_admin_page() {
 		$rendered = apply_filters( 'graphql_render_admin_page', '<div class="wrap"><div id="graphiql" class="graphiql-container">Loading ...</div></div>' );
-		echo $rendered;
+		echo $rendered; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
This should be escaped, since the output can change via a filter.

Reply:
🤔 I think we can use `wp_kses_post` but I wonder if it is too powerful. `wp_kses` would not work since the html could be anything. and using `esc_html` could mess with the html.

Suggestions?!

Reply:
I _think_ `wp_kses_post()` is fine. My only concern is that it strips `data-*` attributes , which probably isnt a common use case for initializing a `GraphiQL` instance, and I wouldn't suggest allowing due to its potential for abuse (i.e. providing config values for a malicious script that gets enqueued somewhere else). And, if a user really needs to use a data attribute, it can still be allowlisted with `add_filter( 'wp_kses_allowed_html' ... )`.

If we want to be extra safe in case some users are _currently_ adding data attributes to the GraphiQL container, we could `phpcs:ignore` for now, and make the change `wp_kses_post` in a separate PR (so it gets its own line-item in the release changelog). 

@jasonbahl ?

Reply:
I think `wp_kses_post()` should work. 

The original intent of this was to allow for plugins (mainly my WPGraphiQL 2 Beta that was in progress at the time) to be able to override the base markup so a different React app could mount to it. 

I think `wp_kses_post()` should allow for custom markup to be output should folks need to do something similar again.

Reply:
Updated with the suggestion! :)